float32[] pipe_direction_options_msg_vector
